docs/walkthrough/using-callbacks.md

```diff
@@ -0,0 +1,128 @@
+            index_data='tll-test-index-da'
+        ),
+        EarlyStopping()
+    ]
+)
+```
+
+## EvaluationCallback
+
+The evaluation callback is used to calculate performance metrics for the model being tuned at the end of each epoch. In order to evaluate the model, two additional data sets - a query dataset and an index dataset - need to be provided as arguments. If no index set is provided, the query dataset is reused instead. Below is an example of the metrics as they are output at the end of finetuning:
+
+```bash
+  Training [5/5] ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 76/76 0:00:00 0:00:16 • loss: 0.003
+[14:10:40] INFO     Done ✨                                                                              __main__.py:194
+           DEBUG    Finetuning took 0 days, 0 hours 3 minutes and 38 seconds                             __main__.py:196
+           DEBUG    Metric: 'resnet50_average_precision' Value: 0.16654                                  __main__.py:205
+           DEBUG    Metric: 'resnet50_dcg_at_k' Value: 0.24018                                           __main__.py:205
+           DEBUG    Metric: 'resnet50_f1_score_at_k' Value: 0.03631                                      __main__.py:205
+           DEBUG    Metric: 'resnet50_hit_at_k' Value: 0.38123                                           __main__.py:205
+           DEBUG    Metric: 'resnet50_ndcg_at_k' Value: 0.24018                                          __main__.py:205
+           DEBUG    Metric: 'resnet50_precision_at_k' Value: 0.01906                                     __main__.py:205
+           DEBUG    Metric: 'resnet50_r_precision' Value: 0.16654                                        __main__.py:205
+           DEBUG    Metric: 'resnet50_recall_at_k' Value: 0.38123                                        __main__.py:205
+           DEBUG    Metric: 'resnet50_reciprocal_rank' Value: 0.16654                                    __main__.py:205
+           INFO     Building the artifact ...                                                            __main__.py:207
+           INFO     Pushing artifact to Hubble ...                                                       __main__.py:231
+```
+
+The evaluation callback is triggered at the end of each epoch, in which the model is evaluated using the `query_data` and `index_data` datasets that were provided when the callback was created.
+It is worth noting that the evaluation callback and the `eval_data` parameter of the fit method do not do the same thing. The eval data parameter takes a `DocumentArray` (or the name of one that has been pushed on Hubble) and uses its contents to evaluate the loss of the model whereas the evaluation callback is used to evaluate the quality of the searches using metrics such as average precision and recall. These search metrics can be used by other callbacks if the evaluation callback is first in the list of callbacks when creating a run.

```

Comment:
let's not mention Hubble, users dont know what Hubble is. You can write cloud instead

Reply:
I've changed this now